### ISSUE TITLE
getJsonConverter().fromJson when the data model changed

### ISSUE THREAD

**Issue by mniedobecki** [javers/javers#768]
hi Javers Team,
we have today faced following issue we. 

Short description of our architecture:

1. In our application we track all changes done by users in the application database. 
2. We call javers.compare for the old and new state of the object to get the Diff object
3. then we call javers.getJsonConverter().toJson(diff) to get the Json format of the diff
4. We store in our audit table the JSON output from the previous call - to track all changes
5. When the user wants to see changes done in the past we load diff Json strings archived in our database, deserialize them to Diff using javers.getJsonConverter().fromJson and then call prettyPrint() for all Diff objects.

This works so far good as long the data model is not changing.

Today we have faced the issue - our data model is under development and is changing and if we use Javers instance initialized with the latest version of the data model and try to convert old Diffs from JSON to Diff object (using fromJson ) it does not work any more - the TypeAdapters used during deserializing the diff JSON try to access the current data model access the properties from the data model classes. If the property which exists in the JSON diff does not exits in the current data model we can not deserialize the diff JSON anymore.

The above behavior might be understandable from the technical point of view but the problem is quite unsolvable - it is not possible to "prettyPrint" old serialized Diffs which have been created in the past without having the exact version of the data model from the past when the diff has been created.

In other words (please correct me if I wrong) but I understand it like - the JSON Diff object is not "complete" or "self-usable" or "self-printable" without having the exact version of the data model. To build the corresponding instance of Diff class the data model is required?

We want to collect the Diff objects (as JSON format) for a very long time period and can not really know know with which exact version of our data model it has been created. Storing the "pretty printed" diff is also not an option - we want to perform customizing of diff printing on the fly dependes on the user preferences.

Is there any solution for this problem?

Many thanks in advance !

**Comment by bartoszwalacik** [comment]
`JsonConverter getJsonConverter()` is not guaranteed to work for old `Changes`. We never said it can be used to deserialize historical `Changes` stored outside of Javers. You are right that it depends on your current model and configuration.

It's just a JSON converter used internally by JaVers and not a stable API of the Javers' persistence framework.

What is guaranteed to be stable is `JaversRepository` which persists `Snapshots` of your domain objects. `Snapshots` should be readable, even after refactoring of your domain classes. 

So my advice is of course, stop persisting `Changes` outside Javers and start using `JaversRepository`.
